In [168]:
import pandas as pd
concat_age1 = pd.read_csv('concat_age1.csv', index_col = 0)
concat_age2 = pd.read_csv('concat_age2.csv', index_col = 0)

C:\Users\User\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3444: DtypeWarning: Columns (45,183,193,200,249,274,374,385,425,568,629,687,692,693,699,701,702,703,704,708,710,712,730,731,738,740,744,745,746,751,783,820,823,842,845,881) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [169]:
# concat_age1 --> 유병여부 컬럼을 제외한 나머지 컬럼들로 구성된 실제 데이터(5개년치)
concat_age1

ID       ID_FAM  REGION  TOWN_T  APT_T      PSU  SEX   AGE  \
0     b'A651228902'  b'A6512289'     1.0     1.0    2.0  b'A651'  2.0  78.0   
1     b'A651417601'  b'A6514176'     1.0     1.0    2.0  b'A651'  1.0  80.0   
2     b'A652228901'  b'A6522289'     1.0     1.0    2.0  b'A652'  1.0  66.0   
3     b'A653184701'  b'A6531847'     1.0     1.0    1.0  b'A653'  2.0  75.0   
4     b'A653235705'  b'A6532357'     1.0     1.0    1.0  b'A653'  2.0  79.0   
...             ...          ...     ...     ...    ...      ...  ...   ...   
8398  b'R804176202'  b'R8041762'     8.0     1.0    2.0  b'R804'  1.0  66.0   
8399  b'R804205101'  b'R8042051'     8.0     1.0    2.0  b'R804'  1.0  75.0   
8400  b'R804205102'  b'R8042051'     8.0     1.0    2.0  b'R804'  2.0  65.0   
8401  b'R804344501'  b'R8043445'     8.0     1.0    2.0  b'R804'  2.0  65.0   
8402  b'R804371701'  b'R8043717'     8.0     1.0    2.0  b'R804'  1.0  65.0   

      INCM  HO_INCM  ...  LF_SECUR_G  AC3_1_03  AC3_2_03  AC8_1_03  BS12_47_4  \
0      3.0      1.0  ...         NaN       NaN       NaN       NaN        NaN   
1      NaN      NaN  ...         NaN       NaN       NaN       NaN        NaN   
2      3.0      3.0  ...         NaN       NaN       NaN       NaN        NaN   
3      2.0      1.0  ...         NaN       NaN       NaN       NaN        NaN   
4      4.0      3.0  ...         NaN       NaN       NaN       NaN        NaN   
...    ...      ...  ...         ...       ...       ...       ...        ...   
8398   2.0      2.0  ...         1.0       NaN       NaN       NaN        NaN   
8399   4.0      2.0  ...         1.0       NaN       NaN       NaN        NaN   
8400   3.0      2.0  ...         1.0       NaN       NaN       NaN        NaN   
8401   2.0      2.0  ...         1.0       NaN       NaN       NaN        NaN   
8402   2.0      2.0  ...         1.0       NaN       NaN       NaN        NaN   

      HE_PLS_30  HE_UCREA_ETC  MO4_17_1  MO4_17_2  T_TYMP_DIS  
0           NaN           NaN       NaN       NaN         NaN  
1           NaN           NaN       NaN       NaN         NaN  
2           NaN           NaN       NaN       NaN         NaN  
3           NaN           NaN       NaN       NaN         NaN  
4           NaN           NaN       NaN       NaN         NaN  
...         ...           ...       ...       ...         ...  
8398       34.0           NaN       1.0       0.0         0.0  
8399       31.0           NaN       0.0       0.0         0.0  
8400       32.0           NaN       NaN       NaN         0.0  
8401       38.0           NaN       1.0       1.0         0.0  
8402        NaN           NaN       0.0       1.0         0.0  

[8403 rows x 884 columns]

In [170]:
# concat_age2 --> 유병여부 컬럼으로 구성된 실제 데이터(5개년치)
concat_age2

ID  DI1_PR  DI2_PR  DI3_PR  DI4_PR  DI5_PR  DM2_PR  DM3_PR  \
0     b'A651228902'     1.0     1.0     NaN     NaN     NaN     1.0     NaN   
1     b'A651417601'     1.0     1.0     NaN     NaN     NaN     NaN     NaN   
2     b'A652228901'     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
3     b'A653184701'     NaN     1.0     NaN     NaN     NaN     NaN     NaN   
4     b'A653235705'     1.0     NaN     NaN     NaN     NaN     NaN     NaN   
...             ...     ...     ...     ...     ...     ...     ...     ...   
8398  b'R804176202'     1.0     1.0     NaN     NaN     NaN     1.0     NaN   
8399  b'R804205101'     0.0     NaN     NaN     NaN     NaN     NaN     NaN   
8400  b'R804205102'     NaN     1.0     NaN     NaN     NaN     NaN     NaN   
8401  b'R804344501'     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
8402  b'R804371701'     NaN     1.0     NaN     NaN     NaN     NaN     NaN   

      DM4_PR  DJ2_PR  ...  DC1_PR  DC2_PR  DC3_PR  DC4_PR  DC5_PR  DC6_PR  \
0        NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
1        NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
2        NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
3        1.0     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
4        NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
...      ...     ...  ...     ...     ...     ...     ...     ...     ...   
8398     NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
8399     NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
8400     NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
8401     NaN     0.0  ...     NaN     NaN     NaN     NaN     NaN     NaN   
8402     NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   

      DC7_PR  DK8_PR  DK9_PR  DK4_PR  
0        NaN     0.0     NaN     NaN  
1        NaN     NaN     NaN     NaN  
2        NaN     NaN     NaN     NaN  
3        NaN     NaN     NaN     NaN  
4        NaN     NaN     NaN     NaN  
...      ...     ...     ...     ...  
8398     NaN     NaN     NaN     NaN  
8399     NaN     NaN     NaN     NaN  
8400     NaN     NaN     NaN     NaN  
8401     NaN     NaN     NaN     NaN  
8402     NaN     NaN     NaN     NaN  

[8403 rows x 29 columns]

In [171]:
# same_code_2 --> 최종 변수설명서
same_code_2 = pd.read_csv('C:/Users/User/VS_code/sas_view/code/same_code_2.csv', index_col = 0)

In [172]:
same_code_2

변수명                   변수설명7  \
0           mod_d               최종 DB 수정일   
1              ID                  개인 아이디   
2          ID_fam                  가구 아이디   
3            year                    조사연도   
4          region                  17개 시도   
..            ...                     ...   
747      T_NQ_OCP           작업장 내 소음노출 경험   
748         HE_DM         당뇨병 유병여부(19세이상)   
749  Total_slp_wk  (만12세이상) 주중 하루 평균 수면시간   
750  Total_slp_wd  (만12세이상) 주말 하루 평균 수면시간   
751      HE_UCREA                  요크레아티닌   

                                               변수설명8  
0                                          최종 DB 수정일  
1                                             개인 아이디  
2                                             가구 아이디  
3                                               조사연도  
4                                             17개 시도  
..                                               ...  
747                                         직업적 소음노출  
748                                  당뇨병 유병여부(19세이상)  
749                (만12세이상) 주중(또는 일하는 날) 하루 평균 수면 시간  
750  (만12세이상) 주말(또는 일하지 않는 날, 일하지 않는 전날) 하루 평균 수면 시간  
751                                           요크레아티닌  

[752 rows x 3 columns]

In [173]:
# col_up --> 변수설명서에 있는 컬럼명을 대문자화 한 것
col_up = []
for i in list(same_code_2['변수명'].values):
    col_up.append(i.upper())

len(col_up)

752

In [174]:
# 변수설명서에는 있지만 실제 데이터(concat_age1)에는 없는 변수들(pr-유병여부 포함)을 제거
for i in ['MOD_D', 'YEAR', 'AGE_MONTH', 'DI1_PR', 'DI2_PR', 'DI4_PR', 'DI5_PR', 'DI6_PR', 'DM3_PR', 'DM4_PR', 'DC1_PR', 'DF2_PR', 'DJ6_PR', 'DH4_PR', 'AC8_1_04', 'AC8_1_05', 'BA1_1', 'BA1_2', 'HE_DPRG', 'HE_WT_PCT', 'HE_BMI_PCT', 'Y_BTH_WT', 'Y_MTM_YN', 'Y_MTM_S1', 'Y_MTM_S2', 'Y_MTM_D1', 'Y_MTM_D2', 'Y_FM_YN', 'Y_FM_S1', 'Y_FM_S2', 'Y_FM_D1', 'Y_FM_D2', 'Y_MLK_ST', 'Y_WN_ST', 'Y_SUP_YN', 'Y_SUP_KD1', 'Y_SUP_KD3', 'Y_SUP_KD4', 'Y_SUP_KD7', 'N_PRG', 'N_BFD_Y', 'N_MEAL', 'N_MEAL_T', 'N_MEAL_P', 'N_MTYPE', 'N_MEAL_W', 'N_DNAME', 'N_DCODE', 'N_CD_VOL', 'N_CD_WT', 'N_TD_VOL', 'N_TD_WT', 'N_FCODE', 'N_FNAME', 'N_ENAME', 'N_FD_NS', 'N_FM_WT', 'N_PRODUCT', 'N_CNAME', 'N_AP', 'N_FCODE3', 'N_FNAME3', 'NF_INTK', 'NF_INTK3', 'DI3_PR', 'DM2_PR', 'DJ2_PR', 'DJ4_PR', 'DE1_PR', 'DE2_PR', 'DC2_PR', 'DC3_PR', 'DC4_PR', 'DC5_PR', 'DC6_PR', 'DC7_PR', 'DL1_PR', 'DJ8_PR', 'DK8_PR', 'DK9_PR', 'DK4_PR', 'LQ4_17', 'LQ4_18', 'LQ4_19', 'LQ4_20', 'AC3_1_04', 'AC3_1_05', 'AC3_2_04', 'AC3_2_05', 'AC8_1E_01', 'AC8_1E_02', 'AC8_1E_03', 'AC8_1E_04', 'AC8_1E_05', 'BS12_45', 'N_KINDG1', 'NF_EN', 'NF_WATER', 'NF_PROT', 'NF_FAT', 'NF_SFA', 'NF_MUFA', 'NF_PUFA', 'NF_N3', 'NF_N6', 'NF_10_0', 'NF_12_0', 'NF_14_0', 'NF_16_0', 'NF_18_0', 'NF_20_0', 'NF_22_0', 'NF_24_0', 'NF_14_1', 'NF_16_1', 'NF_18_1', 'NF_20_1', 'NF_22_1', 'NF_24_1', 'NF_18_2N6', 'NF_18_3N3', 'NF_18_3N6', 'NF_18_4', 'NF_20_2', 'NF_20_3N6', 'NF_20_4N6', 'NF_20_5N3', 'NF_22_2', 'NF_22_5', 'NF_22_6N3', 'NF_CHOL', 'NF_CHO', 'NF_TDF', 'NF_SUGAR', 'NF_CA', 'NF_PHOS', 'NF_FE', 'NF_NA', 'NF_K', 'NF_VA_RAE', 'NF_CAROT', 'NF_RETIN', 'NF_B1', 'NF_B2', 'NF_NIAC', 'NF_FOLATE', 'NF_VITC']:
    col_up.remove(i)
len(col_up)

605

In [175]:
# real_data1 --> 실제데이터(concat_age1)에서 변수설명서에 있는 컬럼들만 뽑아낸 것
# real_data2 --> concat_age2(유병여부 컬럼으로 구성된 실제 데이터)
real_data1 = concat_age1[col_up]
real_data2 = concat_age2

In [176]:
real_data1.to_csv('real_data1.csv', encoding='cp949')
real_data2.to_csv('real_data2.csv', encoding='cp949')

In [177]:
# 실제데이터에 사용된 컬럼은 605개 
# --> 변수설명서에서도 실제로 활용된 컬럼만 남기는 작업 진행
print(col_up)
print(len(col_up))

['ID', 'ID_FAM', 'REGION', 'TOWN_T', 'APT_T', 'PSU', 'SEX', 'AGE', 'INCM', 'HO_INCM', 'INCM5', 'HO_INCM5', 'EDU', 'OCCP', 'WT_HS', 'WT_ITVEX', 'WT_PFT', 'WT_NTR', 'WT_TOT', 'WT_PFNT', 'KSTRATA', 'CFAM', 'GENERTN', 'ALLOWNC', 'LIVE_T', 'AINC', 'MARRI_1', 'MARRI_2', 'FAM_RELA', 'TINS', 'ID_F', 'ID_M', 'D_2_WK', 'DI1_DG', 'DI1_AG', 'DI1_PT', 'DI1_2', 'DI2_AG', 'DI2_2', 'DI3_AG', 'DI3_2', 'DI4_DG', 'DI4_PT', 'DI5_DG', 'DI5_AG', 'DI5_PT', 'DI6_DG', 'DI6_AG', 'DI6_PT', 'DM1_DG', 'DM1_PR', 'DM1_PT', 'DM2_AG', 'DM3_DG', 'DM3_AG', 'DM3_PT', 'DM4_DG', 'DM4_AG', 'DM4_PT', 'DJ2_AG', 'DJ4_DG', 'DJ4_AG', 'DJ4_PT', 'DJ4_3', 'DE1_DG', 'DE1_AG', 'DE1_3', 'DE1_4', 'DE2_AG', 'DC1_DG', 'DC1_AG', 'DC1_PT', 'DC2_DG', 'DC2_AG', 'DC3_DG', 'DC3_AG', 'DC4_DG', 'DC4_AG', 'DC5_DG', 'DC5_AG', 'DC6_DG', 'DC6_AG', 'DC7_DG', 'DC7_AG', 'DC11_DG', 'DC11_AG', 'DC12_DG', 'DC12_AG', 'DF2_AG', 'DL1_DG', 'DL1_AG', 'DJ8_DG', 'DJ8_AG', 'DJ6_DG', 'DJ6_AG', 'DJ6_PT', 'DH4_DG', 'DH4_AG', 'DH4_PT', 'DK8_AG', 'DK9_AG', 'DK4_AG', '

In [178]:
same_code_2

변수명                   변수설명7  \
0           mod_d               최종 DB 수정일   
1              ID                  개인 아이디   
2          ID_fam                  가구 아이디   
3            year                    조사연도   
4          region                  17개 시도   
..            ...                     ...   
747      T_NQ_OCP           작업장 내 소음노출 경험   
748         HE_DM         당뇨병 유병여부(19세이상)   
749  Total_slp_wk  (만12세이상) 주중 하루 평균 수면시간   
750  Total_slp_wd  (만12세이상) 주말 하루 평균 수면시간   
751      HE_UCREA                  요크레아티닌   

                                               변수설명8  
0                                          최종 DB 수정일  
1                                             개인 아이디  
2                                             가구 아이디  
3                                               조사연도  
4                                             17개 시도  
..                                               ...  
747                                         직업적 소음노출  
748                                  당뇨병 유병여부(19세이상)  
749                (만12세이상) 주중(또는 일하는 날) 하루 평균 수면 시간  
750  (만12세이상) 주말(또는 일하지 않는 날, 일하지 않는 전날) 하루 평균 수면 시간  
751                                           요크레아티닌  

[752 rows x 3 columns]

In [179]:
# same_code_2(변수설명서)를 돌면서 
# 실제사용된 컬럼(605개)과 일치하는 컬럼의 인덱스만 추출(index_list)
index_list = []
for i in range(len(same_code_2)):
    if same_code_2.loc[i][0].upper() in col_up:
        index_list.append(i)
index_list

[1,
 2,
 4,
 5,
 6,
 7,
 8,
 9,
 11,
 12,
 13,
 14,
 15,
 16,
 17,
 18,
 19,
 20,
 21,
 22,
 23,
 24,
 25,
 26,
 27,
 28,
 29,
 30,
 31,
 32,
 33,
 34,
 35,
 36,
 37,
 39,
 40,
 41,
 43,
 44,
 45,
 46,
 48,
 49,
 50,
 52,
 53,
 54,
 56,
 57,
 58,
 59,
 60,
 61,
 62,
 64,
 65,
 66,
 68,
 69,
 70,
 71,
 72,
 73,
 74,
 75,
 76,
 77,
 78,
 79,
 80,
 82,
 83,
 84,
 85,
 86,
 87,
 88,
 89,
 90,
 91,
 92,
 93,
 94,
 95,
 96,
 97,
 98,
 99,
 101,
 102,
 103,
 104,
 105,
 106,
 108,
 109,
 110,
 112,
 113,
 114,
 115,
 116,
 117,
 118,
 119,
 120,
 121,
 122,
 123,
 124,
 125,
 126,
 127,
 128,
 129,
 130,
 131,
 132,
 133,
 134,
 135,
 136,
 137,
 138,
 139,
 140,
 141,
 142,
 143,
 144,
 145,
 146,
 147,
 148,
 149,
 150,
 151,
 152,
 155,
 156,
 157,
 158,
 159,
 160,
 161,
 162,
 163,
 164,
 165,
 166,
 167,
 168,
 169,
 170,
 171,
 172,
 173,
 174,
 175,
 176,
 177,
 178,
 179,
 180,
 181,
 182,
 183,
 184,
 185,
 186,
 187,
 188,
 189,
 190,
 191,
 194,
 195,
 196,
 197,
 198,
 199,
 200,

In [180]:
len(index_list)

605

In [181]:
# new_columns --> 변수설명서에서 실제 사용된 컬럼들만 뽑은 것
new_columns = same_code_2.loc[index_list]
new_columns

변수명                   변수설명7  \
1              ID                  개인 아이디   
2          ID_fam                  가구 아이디   
4          region                  17개 시도   
5          town_t                 동/읍면 구분   
6           apt_t                  아파트 구분   
..            ...                     ...   
747      T_NQ_OCP           작업장 내 소음노출 경험   
748         HE_DM         당뇨병 유병여부(19세이상)   
749  Total_slp_wk  (만12세이상) 주중 하루 평균 수면시간   
750  Total_slp_wd  (만12세이상) 주말 하루 평균 수면시간   
751      HE_UCREA                  요크레아티닌   

                                               변수설명8  
1                                             개인 아이디  
2                                             가구 아이디  
4                                             17개 시도  
5                                            동/읍면 구분  
6                                             아파트 구분  
..                                               ...  
747                                         직업적 소음노출  
748                                  당뇨병 유병여부(19세이상)  
749                (만12세이상) 주중(또는 일하는 날) 하루 평균 수면 시간  
750  (만12세이상) 주말(또는 일하지 않는 날, 일하지 않는 전날) 하루 평균 수면 시간  
751                                           요크레아티닌  

[605 rows x 3 columns]

In [182]:
# new_columns(실제 사용된 컬럼만 추출한 변수설명서) 대문자화
upper_columns = []
for i in new_columns['변수명'].values:
    upper_columns.append(i.upper())

upper_columns
new_columns['변수명'] = upper_columns
new_columns = new_columns.reset_index()

In [190]:
new_columns = new_columns.drop(['index'], axis=1)
new_columns

변수명                   변수설명7  \
0              ID                  개인 아이디   
1          ID_FAM                  가구 아이디   
2          REGION                  17개 시도   
3          TOWN_T                 동/읍면 구분   
4           APT_T                  아파트 구분   
..            ...                     ...   
600      T_NQ_OCP           작업장 내 소음노출 경험   
601         HE_DM         당뇨병 유병여부(19세이상)   
602  TOTAL_SLP_WK  (만12세이상) 주중 하루 평균 수면시간   
603  TOTAL_SLP_WD  (만12세이상) 주말 하루 평균 수면시간   
604      HE_UCREA                  요크레아티닌   

                                               변수설명8  
0                                             개인 아이디  
1                                             가구 아이디  
2                                             17개 시도  
3                                            동/읍면 구분  
4                                             아파트 구분  
..                                               ...  
600                                         직업적 소음노출  
601                                  당뇨병 유병여부(19세이상)  
602                (만12세이상) 주중(또는 일하는 날) 하루 평균 수면 시간  
603  (만12세이상) 주말(또는 일하지 않는 날, 일하지 않는 전날) 하루 평균 수면 시간  
604                                           요크레아티닌  

[605 rows x 3 columns]

In [191]:
#  new_columns(실제 사용된 컬럼만 추출한 변수설명서)를 csv파일로 저장
new_columns.to_csv('new_columns.csv', encoding='cp949')

In [184]:
# real_data1 --> 실제데이터(concat_age1)에서 변수설명서에 있는 컬럼들만 뽑아낸 것
# real_data2 --> concat_age2(유병여부 컬럼으로 구성된 실제 데이터)
real_data1

ID       ID_FAM  REGION  TOWN_T  APT_T      PSU  SEX   AGE  \
0     b'A651228902'  b'A6512289'     1.0     1.0    2.0  b'A651'  2.0  78.0   
1     b'A651417601'  b'A6514176'     1.0     1.0    2.0  b'A651'  1.0  80.0   
2     b'A652228901'  b'A6522289'     1.0     1.0    2.0  b'A652'  1.0  66.0   
3     b'A653184701'  b'A6531847'     1.0     1.0    1.0  b'A653'  2.0  75.0   
4     b'A653235705'  b'A6532357'     1.0     1.0    1.0  b'A653'  2.0  79.0   
...             ...          ...     ...     ...    ...      ...  ...   ...   
8398  b'R804176202'  b'R8041762'     8.0     1.0    2.0  b'R804'  1.0  66.0   
8399  b'R804205101'  b'R8042051'     8.0     1.0    2.0  b'R804'  1.0  75.0   
8400  b'R804205102'  b'R8042051'     8.0     1.0    2.0  b'R804'  2.0  65.0   
8401  b'R804344501'  b'R8043445'     8.0     1.0    2.0  b'R804'  2.0  65.0   
8402  b'R804371701'  b'R8043717'     8.0     1.0    2.0  b'R804'  1.0  65.0   

      INCM  HO_INCM  ...  DN1_PT  BS12_1  BS12_2  BS12_37  BS12_47  T_NQ_OCP  \
0      3.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
1      NaN      NaN  ...     NaN     NaN     NaN      NaN      NaN       2.0   
2      3.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
3      2.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       1.0   
4      4.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
...    ...      ...  ...     ...     ...     ...      ...      ...       ...   
8398   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8399   4.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8400   3.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8401   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8402   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       1.0   

      HE_DM  TOTAL_SLP_WK  TOTAL_SLP_WD  HE_UCREA  
0       2.0         480.0         480.0     128.4  
1       2.0         390.0         420.0      34.8  
2       3.0         390.0         390.0     269.8  
3       2.0         360.0         360.0     136.0  
4       3.0         240.0         240.0       NaN  
...     ...           ...           ...       ...  
8398    2.0         480.0         480.0     151.2  
8399    3.0         300.0         300.0      95.5  
8400    2.0         420.0         600.0      35.2  
8401    3.0         480.0         480.0      53.0  
8402    2.0         420.0         420.0      79.7  

[8403 rows x 605 columns]

In [185]:
real_data2

ID  DI1_PR  DI2_PR  DI3_PR  DI4_PR  DI5_PR  DM2_PR  DM3_PR  \
0     b'A651228902'     1.0     1.0     NaN     NaN     NaN     1.0     NaN   
1     b'A651417601'     1.0     1.0     NaN     NaN     NaN     NaN     NaN   
2     b'A652228901'     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
3     b'A653184701'     NaN     1.0     NaN     NaN     NaN     NaN     NaN   
4     b'A653235705'     1.0     NaN     NaN     NaN     NaN     NaN     NaN   
...             ...     ...     ...     ...     ...     ...     ...     ...   
8398  b'R804176202'     1.0     1.0     NaN     NaN     NaN     1.0     NaN   
8399  b'R804205101'     0.0     NaN     NaN     NaN     NaN     NaN     NaN   
8400  b'R804205102'     NaN     1.0     NaN     NaN     NaN     NaN     NaN   
8401  b'R804344501'     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
8402  b'R804371701'     NaN     1.0     NaN     NaN     NaN     NaN     NaN   

      DM4_PR  DJ2_PR  ...  DC1_PR  DC2_PR  DC3_PR  DC4_PR  DC5_PR  DC6_PR  \
0        NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
1        NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
2        NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
3        1.0     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
4        NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
...      ...     ...  ...     ...     ...     ...     ...     ...     ...   
8398     NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
8399     NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
8400     NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
8401     NaN     0.0  ...     NaN     NaN     NaN     NaN     NaN     NaN   
8402     NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   

      DC7_PR  DK8_PR  DK9_PR  DK4_PR  
0        NaN     0.0     NaN     NaN  
1        NaN     NaN     NaN     NaN  
2        NaN     NaN     NaN     NaN  
3        NaN     NaN     NaN     NaN  
4        NaN     NaN     NaN     NaN  
...      ...     ...     ...     ...  
8398     NaN     NaN     NaN     NaN  
8399     NaN     NaN     NaN     NaN  
8400     NaN     NaN     NaN     NaN  
8401     NaN     NaN     NaN     NaN  
8402     NaN     NaN     NaN     NaN  

[8403 rows x 29 columns]

## 실제 데이터(real_data1)의 컬럼 중 값이 전체 다 비어있는 컬럼을 삭제하는 작업

In [1]:
import pandas as pd
real_data1 = pd.read_csv('real_data1.csv', index_col = 0)
real_data2 = pd.read_csv('real_data2.csv', index_col = 0)

c:\Users\User\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3444: DtypeWarning: Columns (31,32,202,265,277,288,291,331,372,399,411,437,443,464,475,482,510,518) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [2]:
real_data1

ID       ID_FAM  REGION  TOWN_T  APT_T      PSU  SEX  AGE  \
0     b'A651228902'  b'A6512289'       1       1      2  b'A651'    2   78   
1     b'A651417601'  b'A6514176'       1       1      2  b'A651'    1   80   
2     b'A652228901'  b'A6522289'       1       1      2  b'A652'    1   66   
3     b'A653184701'  b'A6531847'       1       1      1  b'A653'    2   75   
4     b'A653235705'  b'A6532357'       1       1      1  b'A653'    2   79   
...             ...          ...     ...     ...    ...      ...  ...  ...   
8398  b'R804176202'  b'R8041762'       8       1      2  b'R804'    1   66   
8399  b'R804205101'  b'R8042051'       8       1      2  b'R804'    1   75   
8400  b'R804205102'  b'R8042051'       8       1      2  b'R804'    2   65   
8401  b'R804344501'  b'R8043445'       8       1      2  b'R804'    2   65   
8402  b'R804371701'  b'R8043717'       8       1      2  b'R804'    1   65   

      INCM  HO_INCM  ...  DN1_PT  BS12_1  BS12_2  BS12_37  BS12_47  T_NQ_OCP  \
0      3.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
1      NaN      NaN  ...     NaN     NaN     NaN      NaN      NaN       2.0   
2      3.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
3      2.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       1.0   
4      4.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
...    ...      ...  ...     ...     ...     ...      ...      ...       ...   
8398   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8399   4.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8400   3.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8401   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8402   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       1.0   

      HE_DM  TOTAL_SLP_WK  TOTAL_SLP_WD  HE_UCREA  
0       2.0         480.0         480.0     128.4  
1       2.0         390.0         420.0      34.8  
2       3.0         390.0         390.0     269.8  
3       2.0         360.0         360.0     136.0  
4       3.0         240.0         240.0       NaN  
...     ...           ...           ...       ...  
8398    2.0         480.0         480.0     151.2  
8399    3.0         300.0         300.0      95.5  
8400    2.0         420.0         600.0      35.2  
8401    3.0         480.0         480.0      53.0  
8402    2.0         420.0         420.0      79.7  

[8403 rows x 605 columns]

In [3]:
real_data2

ID  DI1_PR  DI2_PR  DI3_PR  DI4_PR  DI5_PR  DM2_PR  DM3_PR  \
0     b'A651228902'     1.0     1.0     NaN     NaN     NaN     1.0     NaN   
1     b'A651417601'     1.0     1.0     NaN     NaN     NaN     NaN     NaN   
2     b'A652228901'     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
3     b'A653184701'     NaN     1.0     NaN     NaN     NaN     NaN     NaN   
4     b'A653235705'     1.0     NaN     NaN     NaN     NaN     NaN     NaN   
...             ...     ...     ...     ...     ...     ...     ...     ...   
8398  b'R804176202'     1.0     1.0     NaN     NaN     NaN     1.0     NaN   
8399  b'R804205101'     0.0     NaN     NaN     NaN     NaN     NaN     NaN   
8400  b'R804205102'     NaN     1.0     NaN     NaN     NaN     NaN     NaN   
8401  b'R804344501'     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
8402  b'R804371701'     NaN     1.0     NaN     NaN     NaN     NaN     NaN   

      DM4_PR  DJ2_PR  ...  DC1_PR  DC2_PR  DC3_PR  DC4_PR  DC5_PR  DC6_PR  \
0        NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
1        NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
2        NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
3        1.0     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
4        NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
...      ...     ...  ...     ...     ...     ...     ...     ...     ...   
8398     NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
8399     NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
8400     NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   
8401     NaN     0.0  ...     NaN     NaN     NaN     NaN     NaN     NaN   
8402     NaN     NaN  ...     NaN     NaN     NaN     NaN     NaN     NaN   

      DC7_PR  DK8_PR  DK9_PR  DK4_PR  
0        NaN     0.0     NaN     NaN  
1        NaN     NaN     NaN     NaN  
2        NaN     NaN     NaN     NaN  
3        NaN     NaN     NaN     NaN  
4        NaN     NaN     NaN     NaN  
...      ...     ...     ...     ...  
8398     NaN     NaN     NaN     NaN  
8399     NaN     NaN     NaN     NaN  
8400     NaN     NaN     NaN     NaN  
8401     NaN     NaN     NaN     NaN  
8402     NaN     NaN     NaN     NaN  

[8403 rows x 29 columns]

In [10]:
# real_data1에서 결측치가 8403인 컬럼 즉 컬럼 전체가 비어있는 컬럼을 추출
real_data1.columns[real_data1.isnull().sum()==8403]

Index(['DF1_YD', 'DF1_YA', 'DJ9_YD', 'DJ9_YA', 'CH2_1', 'CH2_2', 'BS10_1',
       'BS2', 'BS10_2', 'BS10_3', 'BE9'],
      dtype='object')

In [11]:
real_data1 = pd.read_csv('real_data1.csv', index_col = 0)
real_data1

c:\Users\User\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3444: DtypeWarning: Columns (31,32,202,265,277,288,291,331,372,399,411,437,443,464,475,482,510,518) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


ID       ID_FAM  REGION  TOWN_T  APT_T      PSU  SEX  AGE  \
0     b'A651228902'  b'A6512289'       1       1      2  b'A651'    2   78   
1     b'A651417601'  b'A6514176'       1       1      2  b'A651'    1   80   
2     b'A652228901'  b'A6522289'       1       1      2  b'A652'    1   66   
3     b'A653184701'  b'A6531847'       1       1      1  b'A653'    2   75   
4     b'A653235705'  b'A6532357'       1       1      1  b'A653'    2   79   
...             ...          ...     ...     ...    ...      ...  ...  ...   
8398  b'R804176202'  b'R8041762'       8       1      2  b'R804'    1   66   
8399  b'R804205101'  b'R8042051'       8       1      2  b'R804'    1   75   
8400  b'R804205102'  b'R8042051'       8       1      2  b'R804'    2   65   
8401  b'R804344501'  b'R8043445'       8       1      2  b'R804'    2   65   
8402  b'R804371701'  b'R8043717'       8       1      2  b'R804'    1   65   

      INCM  HO_INCM  ...  DN1_PT  BS12_1  BS12_2  BS12_37  BS12_47  T_NQ_OCP  \
0      3.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
1      NaN      NaN  ...     NaN     NaN     NaN      NaN      NaN       2.0   
2      3.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
3      2.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       1.0   
4      4.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
...    ...      ...  ...     ...     ...     ...      ...      ...       ...   
8398   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8399   4.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8400   3.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8401   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8402   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       1.0   

      HE_DM  TOTAL_SLP_WK  TOTAL_SLP_WD  HE_UCREA  
0       2.0         480.0         480.0     128.4  
1       2.0         390.0         420.0      34.8  
2       3.0         390.0         390.0     269.8  
3       2.0         360.0         360.0     136.0  
4       3.0         240.0         240.0       NaN  
...     ...           ...           ...       ...  
8398    2.0         480.0         480.0     151.2  
8399    3.0         300.0         300.0      95.5  
8400    2.0         420.0         600.0      35.2  
8401    3.0         480.0         480.0      53.0  
8402    2.0         420.0         420.0      79.7  

[8403 rows x 605 columns]

In [13]:
real_data1.to_csv('real_data1.csv', encoding='cp949')

In [14]:
# 실제 데이터(real_data1)의 컬럼 중 값이 전체 다 비어있는 컬럼(11개)을 삭제하는 작업
real_data1.drop(['DF1_YD', 'DF1_YA', 'DJ9_YD', 'DJ9_YA', 'CH2_1', 'CH2_2', 'BS10_1',
       'BS2', 'BS10_2', 'BS10_3', 'BE9'], axis=1, inplace=True)

In [21]:
# 전체가 비어 있는 컬럼을 삭제한 실제 데이터(real_data1)을 real_data1_notnull.csv파일로 저장
real_data1.to_csv('real_data1_notnull.csv', encoding='cp949')

In [5]:
import pandas as pd

real_data1 = pd.read_csv('real_data1_notnull.csv', encoding='cp949', index_col=0)
real_data1

c:\Users\User\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3444: DtypeWarning: Columns (31,32,202,265,277,288,291,331,372,399,411,433,439,458,469,476,504,512) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


ID       ID_FAM  REGION  TOWN_T  APT_T      PSU  SEX  AGE  \
0     b'A651228902'  b'A6512289'       1       1      2  b'A651'    2   78   
1     b'A651417601'  b'A6514176'       1       1      2  b'A651'    1   80   
2     b'A652228901'  b'A6522289'       1       1      2  b'A652'    1   66   
3     b'A653184701'  b'A6531847'       1       1      1  b'A653'    2   75   
4     b'A653235705'  b'A6532357'       1       1      1  b'A653'    2   79   
...             ...          ...     ...     ...    ...      ...  ...  ...   
8398  b'R804176202'  b'R8041762'       8       1      2  b'R804'    1   66   
8399  b'R804205101'  b'R8042051'       8       1      2  b'R804'    1   75   
8400  b'R804205102'  b'R8042051'       8       1      2  b'R804'    2   65   
8401  b'R804344501'  b'R8043445'       8       1      2  b'R804'    2   65   
8402  b'R804371701'  b'R8043717'       8       1      2  b'R804'    1   65   

      INCM  HO_INCM  ...  DN1_PT  BS12_1  BS12_2  BS12_37  BS12_47  T_NQ_OCP  \
0      3.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
1      NaN      NaN  ...     NaN     NaN     NaN      NaN      NaN       2.0   
2      3.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
3      2.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       1.0   
4      4.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
...    ...      ...  ...     ...     ...     ...      ...      ...       ...   
8398   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8399   4.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8400   3.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8401   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8402   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       1.0   

      HE_DM  TOTAL_SLP_WK  TOTAL_SLP_WD  HE_UCREA  
0       2.0         480.0         480.0     128.4  
1       2.0         390.0         420.0      34.8  
2       3.0         390.0         390.0     269.8  
3       2.0         360.0         360.0     136.0  
4       3.0         240.0         240.0       NaN  
...     ...           ...           ...       ...  
8398    2.0         480.0         480.0     151.2  
8399    3.0         300.0         300.0      95.5  
8400    2.0         420.0         600.0      35.2  
8401    3.0         480.0         480.0      53.0  
8402    2.0         420.0         420.0      79.7  

[8403 rows x 594 columns]

In [8]:
# real_data1의 ID컬럼에는 중복된 값 존재X
len(real_data1['ID'].unique())

8403

### <N_EN>은 1일 에너지 섭취량
### <NF_EN>은 에너지 섭취량인데 실제 데이터에 해당 컬럼 존재X

In [18]:
real_data1['N_EN']

0               NaN
1       1677.250403
2       2907.349349
3        689.726391
4        821.963207
           ...     
8398    1561.599533
8399    1541.109615
8400    2276.859154
8401    1469.044167
8402    2953.040528
Name: N_EN, Length: 8403, dtype: float64

In [23]:
# 영양과 관련된(N_ 이 포함된) 컬럼들
real_data1[['N_DIET', 'N_DIET_WHY', 'N_DT_DS', 'N_DT_ETC', 'N_DUSUAL',
       'N_WAT_C', 'N_DAY', 'N_INTK', 'N_EN', 'N_WATER', 'N_PROT', 'N_FAT',
       'N_SFA', 'N_MUFA', 'N_PUFA', 'N_N3', 'N_N6', 'N_CHOL', 'N_CHO', 'N_TDF',
       'N_SUGAR', 'N_CA', 'N_PHOS', 'N_FE', 'N_NA', 'N_K', 'N_VA_RAE',
       'N_CAROT', 'N_RETIN', 'N_B1', 'N_B2', 'N_NIAC', 'N_FOLATE', 'N_VITC']]

N_DIET  N_DIET_WHY                              N_DT_DS  \
0        NaN         NaN                                  NaN   
1        1.0         3.0                                  NaN   
2        2.0         NaN                                  NaN   
3        1.0         1.0  b'\xc0\xa7\xc0\xe5\xc0\xe5\xbe\xd6'   
4        2.0         NaN                                  NaN   
...      ...         ...                                  ...   
8398     1.0         2.0                                  NaN   
8399     1.0         1.0                  b'\xb4\xe7\xb4\xa2'   
8400     2.0         NaN                                  NaN   
8401     1.0         2.0                                  NaN   
8402     2.0         NaN                                  NaN   

                                 N_DT_ETC N_DUSUAL  N_WAT_C  \
0                                     NaN      NaN      NaN   
1     b'\xb0\xc7\xb0\xad\xb0\xfc\xb8\xae'     b'2'      3.8   
2                                     NaN     b'1'      3.0   
3                                     NaN     b'2'      6.0   
4                                     NaN     b'3'      2.0   
...                                   ...      ...      ...   
8398                                  NaN     b'2'      6.7   
8399                                  NaN     b'2'      7.5   
8400                                  NaN     b'2'      5.0   
8401                                  NaN     b'2'      2.0   
8402                                  NaN     b'2'      5.0   

                            N_DAY       N_INTK         N_EN      N_WATER  ...  \
0                             NaN          NaN          NaN          NaN  ...   
1     b'\xb8\xf1\xbf\xe4\xc0\xcf'  1544.317922  1677.250403   972.254662  ...   
2     b'\xbc\xf6\xbf\xe4\xc0\xcf'  1271.523593  2907.349349   659.077288  ...   
3     b'\xbc\xf6\xbf\xe4\xc0\xcf'   324.185945   689.726391   162.617404  ...   
4     b'\xc8\xad\xbf\xe4\xc0\xcf'   525.681802   821.963207   332.126544  ...   
...                           ...          ...          ...          ...  ...   
8398  b'\xb8\xf1\xbf\xe4\xc0\xcf'  1690.975469  1561.599533  1322.946789  ...   
8399  b'\xb8\xf1\xbf\xe4\xc0\xcf'  1192.137456  1541.109615   832.900707  ...   
8400  b'\xb8\xf1\xbf\xe4\xc0\xcf'  1442.338766  2276.859154   925.911409  ...   
8401  b'\xb1\xdd\xbf\xe4\xc0\xcf'  1577.858728  1469.044167  1112.335133  ...   
8402  b'\xb8\xf1\xbf\xe4\xc0\xcf'  1972.433348  2953.040528   903.365806  ...   

             N_NA          N_K    N_VA_RAE      N_CAROT     N_RETIN      N_B1  \
0             NaN          NaN         NaN          NaN         NaN       NaN   
1     4399.799642  2566.620279  294.793594   998.477352  211.587148  1.006231   
2     3929.082290  2331.404861  163.018429  1800.234676   12.998873  1.624364   
3     1157.897389   530.823816  186.218018  1385.932537   70.723640  0.356052   
4     1202.976063   647.614546   27.905785   279.510000    4.613285  0.280312   
...           ...          ...         ...          ...         ...       ...   
8398  2009.578556  2644.914837  218.657333   872.716548  145.930954  1.068906   
8399  2936.950002  3483.531554  291.119212  1821.338545  139.341000  1.103136   
8400  2799.353637  3213.265399  358.105544  1646.836002  220.869211  1.206207   
8401  3037.835581  1991.461975  286.170166  1229.644393  183.699800  0.625029   
8402  3681.501097  3665.578194  144.793359  1593.550499   11.997484  1.774904   

          N_B2     N_NIAC    N_FOLATE      N_VITC  
0          NaN        NaN         NaN         NaN  
1     1.732103   8.271924  386.525342  101.079339  
2     1.005462  13.698492  217.558630   15.821594  
3     0.889463   2.004974  119.056346   25.218942  
4     0.227525   3.437222  112.559393    8.250554  
...        ...        ...         ...         ...  
8398  1.267011  12.471559  239.418558   29.636101  
8399  2.078078  16.175646  647.037982   46.567949  
8400  1.913871  13.351069  506.184039   44.060558  
84

In [39]:
new_columns = pd.read_csv('new_columns.csv', encoding='cp949', index_col=0)
new_columns

변수명                   변수설명7  \
0              ID                  개인 아이디   
1          ID_FAM                  가구 아이디   
2          REGION                  17개 시도   
3          TOWN_T                 동/읍면 구분   
4           APT_T                  아파트 구분   
..            ...                     ...   
600      T_NQ_OCP           작업장 내 소음노출 경험   
601         HE_DM         당뇨병 유병여부(19세이상)   
602  TOTAL_SLP_WK  (만12세이상) 주중 하루 평균 수면시간   
603  TOTAL_SLP_WD  (만12세이상) 주말 하루 평균 수면시간   
604      HE_UCREA                  요크레아티닌   

                                               변수설명8              세부정보  
0                                             개인 아이디               NaN  
1                                             가구 아이디               NaN  
2                                             17개 시도               NaN  
3                                            동/읍면 구분               NaN  
4                                             아파트 구분               NaN  
..                                               ...               ...  
600                                         직업적 소음노출         1=예/2=아니오  
601                                  당뇨병 유병여부(19세이상)  1=정상/2=전단계/3=당뇨병  
602                (만12세이상) 주중(또는 일하는 날) 하루 평균 수면 시간               NaN  
603  (만12세이상) 주말(또는 일하지 않는 날, 일하지 않는 전날) 하루 평균 수면 시간               NaN  
604                                           요크레아티닌               NaN  

[605 rows x 4 columns]

In [48]:
# 변수설명7 or 8 에 '사유'를 포함하고 있는 컬럼들(34개)
sayu_col = list(new_columns['변수명'][new_columns['변수설명7'].str.contains('사유')])
print(sayu_col)
print(len(sayu_col))

['LQ4_01', 'LQ4_02', 'LQ4_03', 'LQ4_04', 'LQ4_05', 'LQ4_06', 'LQ4_07', 'LQ4_08', 'LQ4_09', 'LQ4_10', 'LQ4_11', 'LQ4_12', 'LQ4_13', 'LQ4_14', 'LQ4_15', 'LQ4_16', 'LQ4_21', 'LQ4_22', 'LQ4_25', 'LQ4_26', 'LQ4_27', 'LQ4_28', 'LQ4_29', 'LQ4_23', 'LW_MP_E', 'BM13_1', 'BM13_2', 'BM13_3', 'BM13_4', 'BM13_5', 'BM13_6', 'M_2_RS', 'M_2_ET', 'LQ4_24']
34


In [47]:
real_data1

ID       ID_FAM  REGION  TOWN_T  APT_T      PSU  SEX  AGE  \
0     b'A651228902'  b'A6512289'       1       1      2  b'A651'    2   78   
1     b'A651417601'  b'A6514176'       1       1      2  b'A651'    1   80   
2     b'A652228901'  b'A6522289'       1       1      2  b'A652'    1   66   
3     b'A653184701'  b'A6531847'       1       1      1  b'A653'    2   75   
4     b'A653235705'  b'A6532357'       1       1      1  b'A653'    2   79   
...             ...          ...     ...     ...    ...      ...  ...  ...   
8398  b'R804176202'  b'R8041762'       8       1      2  b'R804'    1   66   
8399  b'R804205101'  b'R8042051'       8       1      2  b'R804'    1   75   
8400  b'R804205102'  b'R8042051'       8       1      2  b'R804'    2   65   
8401  b'R804344501'  b'R8043445'       8       1      2  b'R804'    2   65   
8402  b'R804371701'  b'R8043717'       8       1      2  b'R804'    1   65   

      INCM  HO_INCM  ...  DN1_PT  BS12_1  BS12_2  BS12_37  BS12_47  T_NQ_OCP  \
0      3.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
1      NaN      NaN  ...     NaN     NaN     NaN      NaN      NaN       2.0   
2      3.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
3      2.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       1.0   
4      4.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
...    ...      ...  ...     ...     ...     ...      ...      ...       ...   
8398   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8399   4.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8400   3.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8401   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8402   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       1.0   

      HE_DM  TOTAL_SLP_WK  TOTAL_SLP_WD  HE_UCREA  
0       2.0         480.0         480.0     128.4  
1       2.0         390.0         420.0      34.8  
2       3.0         390.0         390.0     269.8  
3       2.0         360.0         360.0     136.0  
4       3.0         240.0         240.0       NaN  
...     ...           ...           ...       ...  
8398    2.0         480.0         480.0     151.2  
8399    3.0         300.0         300.0      95.5  
8400    2.0         420.0         600.0      35.2  
8401    3.0         480.0         480.0      53.0  
8402    2.0         420.0         420.0      79.7  

[8403 rows x 594 columns]

In [49]:
# 변수설명7 or 8 에 '사유'를 포함하고 있는 컬럼들 삭제
real_data1.drop(sayu_col, axis=1, inplace=True)
real_data1

ID       ID_FAM  REGION  TOWN_T  APT_T      PSU  SEX  AGE  \
0     b'A651228902'  b'A6512289'       1       1      2  b'A651'    2   78   
1     b'A651417601'  b'A6514176'       1       1      2  b'A651'    1   80   
2     b'A652228901'  b'A6522289'       1       1      2  b'A652'    1   66   
3     b'A653184701'  b'A6531847'       1       1      1  b'A653'    2   75   
4     b'A653235705'  b'A6532357'       1       1      1  b'A653'    2   79   
...             ...          ...     ...     ...    ...      ...  ...  ...   
8398  b'R804176202'  b'R8041762'       8       1      2  b'R804'    1   66   
8399  b'R804205101'  b'R8042051'       8       1      2  b'R804'    1   75   
8400  b'R804205102'  b'R8042051'       8       1      2  b'R804'    2   65   
8401  b'R804344501'  b'R8043445'       8       1      2  b'R804'    2   65   
8402  b'R804371701'  b'R8043717'       8       1      2  b'R804'    1   65   

      INCM  HO_INCM  ...  DN1_PT  BS12_1  BS12_2  BS12_37  BS12_47  T_NQ_OCP  \
0      3.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
1      NaN      NaN  ...     NaN     NaN     NaN      NaN      NaN       2.0   
2      3.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
3      2.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       1.0   
4      4.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
...    ...      ...  ...     ...     ...     ...      ...      ...       ...   
8398   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8399   4.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8400   3.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8401   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8402   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       1.0   

      HE_DM  TOTAL_SLP_WK  TOTAL_SLP_WD  HE_UCREA  
0       2.0         480.0         480.0     128.4  
1       2.0         390.0         420.0      34.8  
2       3.0         390.0         390.0     269.8  
3       2.0         360.0         360.0     136.0  
4       3.0         240.0         240.0       NaN  
...     ...           ...           ...       ...  
8398    2.0         480.0         480.0     151.2  
8399    3.0         300.0         300.0      95.5  
8400    2.0         420.0         600.0      35.2  
8401    3.0         480.0         480.0      53.0  
8402    2.0         420.0         420.0      79.7  

[8403 rows x 560 columns]

In [50]:
real_data1.to_csv('real_data1_no_sayu.csv', encoding='cp949')

In [52]:
# 값이 이상한 컬럼들 삭제(11개)
real_data1.drop(['HE_UCOT_ETC','HE_UKAL_ETC','BM14_2',
'N_DT_DS','N_DT_ETC','N_DUSUAL','N_DAY','DC11_TP','BH1_7',
'BH2_65','BS5_31'], axis=1, inplace=True)

In [53]:
real_data1

ID       ID_FAM  REGION  TOWN_T  APT_T      PSU  SEX  AGE  \
0     b'A651228902'  b'A6512289'       1       1      2  b'A651'    2   78   
1     b'A651417601'  b'A6514176'       1       1      2  b'A651'    1   80   
2     b'A652228901'  b'A6522289'       1       1      2  b'A652'    1   66   
3     b'A653184701'  b'A6531847'       1       1      1  b'A653'    2   75   
4     b'A653235705'  b'A6532357'       1       1      1  b'A653'    2   79   
...             ...          ...     ...     ...    ...      ...  ...  ...   
8398  b'R804176202'  b'R8041762'       8       1      2  b'R804'    1   66   
8399  b'R804205101'  b'R8042051'       8       1      2  b'R804'    1   75   
8400  b'R804205102'  b'R8042051'       8       1      2  b'R804'    2   65   
8401  b'R804344501'  b'R8043445'       8       1      2  b'R804'    2   65   
8402  b'R804371701'  b'R8043717'       8       1      2  b'R804'    1   65   

      INCM  HO_INCM  ...  DN1_PT  BS12_1  BS12_2  BS12_37  BS12_47  T_NQ_OCP  \
0      3.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
1      NaN      NaN  ...     NaN     NaN     NaN      NaN      NaN       2.0   
2      3.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
3      2.0      1.0  ...     NaN     2.0     NaN      NaN      NaN       1.0   
4      4.0      3.0  ...     NaN     2.0     NaN      NaN      NaN       2.0   
...    ...      ...  ...     ...     ...     ...      ...      ...       ...   
8398   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8399   4.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8400   3.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8401   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       2.0   
8402   2.0      2.0  ...     NaN     2.0     NaN      2.0      NaN       1.0   

      HE_DM  TOTAL_SLP_WK  TOTAL_SLP_WD  HE_UCREA  
0       2.0         480.0         480.0     128.4  
1       2.0         390.0         420.0      34.8  
2       3.0         390.0         390.0     269.8  
3       2.0         360.0         360.0     136.0  
4       3.0         240.0         240.0       NaN  
...     ...           ...           ...       ...  
8398    2.0         480.0         480.0     151.2  
8399    3.0         300.0         300.0      95.5  
8400    2.0         420.0         600.0      35.2  
8401    3.0         480.0         480.0      53.0  
8402    2.0         420.0         420.0      79.7  

[8403 rows x 549 columns]

In [54]:
real_data1.to_csv('real_data1_final.csv', encoding='cp949')

## 5/10까지 전처리 완료한 실제 데이터(real_data1_final.csv)  --> 컬럼 549개